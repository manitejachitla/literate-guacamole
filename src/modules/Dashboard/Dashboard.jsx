import React from 'react'
import { Container, Row, Col, Button } from 'reactstrap';

import { TodoStatus } from '../../config/const'
import CardInfo from '../../shared/CardInfo/CardInfo';
import ToolInfo from '../../shared/ToolInfo/ToolInfo';

const Dashboard = ({
    setCreateTodoModal,
    todoStatus,
    inProgressStatus,
    inQAStatus,
    doneStatus,
    blockedStatus,
    deployedStatus
}) => {

    return (
        <Container fluid className="dashboard_section">
            <Row className="mb-3">
                <Col>
                    <Button className="create_todo_btn" onClick={setCreateTodoModal}>
                        Create Todo
                    </Button>
                </Col>
            </Row>
            <Row>
                <Col xs={12} md={3} lg={2} className="mb-4">
                    <ToolInfo title={TodoStatus[1]}>
                        {todoStatus && todoStatus.map((todo) => (
                            <CardInfo todo={todo} id={todo.id} title={todo.title} description={todo.description} openTodoModal={(e)=>setCreateTodoModal(e)}/>
                        ))}
                    </ToolInfo>
                </Col>
                <Col xs={12} md={3} lg={2} className="mb-4">
                    <ToolInfo title={TodoStatus[2]}>
                        {inProgressStatus && inProgressStatus.map((todo) => (
                            <CardInfo todo={todo} id={todo.id} title={todo.title} description={todo.description} openTodoModal={(e)=>setCreateTodoModal(e)} />
                        ))}
                    </ToolInfo>
                </Col>
                <Col xs={12} md={3} lg={2}className="mb-4">
                    <ToolInfo title={TodoStatus[3]}>
                        {inQAStatus && inQAStatus.map((todo) => (
                            <CardInfo todo={todo} id={todo.id} title={todo.title} description={todo.description} openTodoModal={(e)=>setCreateTodoModal(e)} />
                        ))}
                    </ToolInfo>
                </Col>
                <Col xs={12} md={3} lg={2} className="mb-4">
                    <ToolInfo title={TodoStatus[4]}>
                        {doneStatus && doneStatus.map((todo) => (
                            <CardInfo todo={todo} id={todo.id} title={todo.title} description={todo.description} openTodoModal={(e)=>setCreateTodoModal(e)} />
                        ))}
                    </ToolInfo>
                </Col>
                <Col xs={12} md={3} lg={2} className="mb-4">
                    <ToolInfo title={TodoStatus[5]}>
                        {blockedStatus && blockedStatus.map((todo) => (
                            <CardInfo todo={todo} id={todo.id} title={todo.title} description={todo.description} openTodoModal={(e)=>setCreateTodoModal(e)} />
                        ))}
                    </ToolInfo>
                </Col>
                <Col xs={12} md={3} lg={2} className="mb-4">
                    <ToolInfo title={TodoStatus[6]}>
                        {deployedStatus && deployedStatus.map((todo) => (
                            <CardInfo todo={todo} id={todo.id} title={todo.title} description={todo.description} openTodoModal={(e)=>setCreateTodoModal(e)} />
                        ))}
                    </ToolInfo>
                </Col>
            </Row>
        </Container>
    )
}

export default Dashboard