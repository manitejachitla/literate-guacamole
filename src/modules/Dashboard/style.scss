.dashboard_section {
    width: 100%;
    height: 100%;
    background: linear-gradient(180deg, #42548e 0%, #6b668c 100%);
    padding-top: 15px;
    padding-bottom: 15px;
    .create_todo_btn {
        background-color: #6c7bab;
        border: unset;
        font-weight: 500;
        &:hover {
            background-color: #6c7bab;
            border: unset;
            border-color: unset;
        }
        &:active {
            background-color: #6c7bab!important;
            border: unset;
            border-color: unset;
        }
        &:focus {
            background-color: #6c7bab!important;
            border: unset;
            box-shadow: unset!important;
            border-color: unset;
        }
    }
}