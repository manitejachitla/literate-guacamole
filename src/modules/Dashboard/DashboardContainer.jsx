import React from 'react'

import Dashboard from './Dashboard'
import CreateTodoModal from '../../shared/CreateTodoModal/CreateTodoModal'
import {
    useTodoState,
    setCreateTodoModal
} from '../../context/TodoContext';

import './style.scss';

const DashboardContainer = () => {
    const todoContext = useTodoState()
    const todoState = todoContext.todoState;

    return (
        <>
            <Dashboard
                todoStatus={todoState.list.filter((list) => list.status === 1)}
                inProgressStatus={todoState.list.filter((list) => list.status === 2)}
                inQAStatus={todoState.list.filter((list) => list.status === 3)}
                doneStatus={todoState.list.filter((list) => list.status === 4)}
                blockedStatus={todoState.list.filter((list) => list.status === 5)}
                deployedStatus={todoState.list.filter((list) => list.status === 6)}
                setCreateTodoModal={(e) => setCreateTodoModal(todoContext.setTodoState, { status: true, data: e })}
            />
            <CreateTodoModal />
        </>
    )
}

export default DashboardContainer