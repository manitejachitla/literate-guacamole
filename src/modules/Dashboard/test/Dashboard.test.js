import Dashboard from '../Dashboard';
import { create } from 'react-test-renderer'

const todoProps = {
  todoStatus: [{ id: 1, title: 'test1', description: 'description1', status: 1, assignee: 'test1' }],
  inProgressStatus: [{ id: 2, title: 'test2', description: 'description2', status: 2, assignee: 'test2' }],
  inQAStatus: [{ id: 3, title: 'test3', description: 'description3', status: 3, assignee: 'test3' }],
  doneStatus: [{ id: 4, title: 'test4', description: 'description4', status: 4, assignee: 'test4' }],
  blockedStatus: [{ id: 5, title: 'test5', description: 'description5', status: 5, assignee: 'test5' }],
  deployedStatus: [{ id: 6, title: 'test6', description: 'description6', status: 6, assignee: 'test6' }],
  setCreateTodoModal: jest.fn()
}


describe('ActivityDetails component', () => {
  test('Matches the snapshot', () => {
    const DashboardComponent = create(
      <Dashboard {...todoProps} />
    )
    expect(DashboardComponent.toJSON()).toMatchSnapshot()
  })
})
