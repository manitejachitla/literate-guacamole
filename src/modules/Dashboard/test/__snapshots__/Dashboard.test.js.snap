// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityDetails component Matches the snapshot 1`] = `
<div
  className="dashboard_section container-fluid"
>
  <div
    className="mb-3 row"
  >
    <div
      className="col"
    >
      <button
        aria-label={null}
        className="create_todo_btn btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        Create Todo
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="mb-4 col-12 col-md-3 col-lg-2"
    >
      <div
        className="tool_info card"
      >
        <div
          className="tool_into_title card-title"
        >
          ToDo
        </div>
        <div
          className="p-0 card-body"
        >
          <div
            className="card_info_section card"
            onClick={[Function]}
          >
            <div
              className="card_info_title card-title"
            >
              test1
            </div>
            <div
              className="p-0 card-body"
            >
              <p
                className="card_info_description"
              >
                description1
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mb-4 col-12 col-md-3 col-lg-2"
    >
      <div
        className="tool_info card"
      >
        <div
          className="tool_into_title card-title"
        >
          InProgress
        </div>
        <div
          className="p-0 card-body"
        >
          <div
            className="card_info_section card"
            onClick={[Function]}
          >
            <div
              className="card_info_title card-title"
            >
              test2
            </div>
            <div
              className="p-0 card-body"
            >
              <p
                className="card_info_description"
              >
                description2
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mb-4 col-12 col-md-3 col-lg-2"
    >
      <div
        className="tool_info card"
      >
        <div
          className="tool_into_title card-title"
        >
          InQA
        </div>
        <div
          className="p-0 card-body"
        >
          <div
            className="card_info_section card"
            onClick={[Function]}
          >
            <div
              className="card_info_title card-title"
            >
              test3
            </div>
            <div
              className="p-0 card-body"
            >
              <p
                className="card_info_description"
              >
                description3
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mb-4 col-12 col-md-3 col-lg-2"
    >
      <div
        className="tool_info card"
      >
        <div
          className="tool_into_title card-title"
        >
          Done
        </div>
        <div
          className="p-0 card-body"
        >
          <div
            className="card_info_section card"
            onClick={[Function]}
          >
            <div
              className="card_info_title card-title"
            >
              test4
            </div>
            <div
              className="p-0 card-body"
            >
              <p
                className="card_info_description"
              >
                description4
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mb-4 col-12 col-md-3 col-lg-2"
    >
      <div
        className="tool_info card"
      >
        <div
          className="tool_into_title card-title"
        >
          Blocked
        </div>
        <div
          className="p-0 card-body"
        >
          <div
            className="card_info_section card"
            onClick={[Function]}
          >
            <div
              className="card_info_title card-title"
            >
              test5
            </div>
            <div
              className="p-0 card-body"
            >
              <p
                className="card_info_description"
              >
                description5
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="mb-4 col-12 col-md-3 col-lg-2"
    >
      <div
        className="tool_info card"
      >
        <div
          className="tool_into_title card-title"
        >
          Deployed
        </div>
        <div
          className="p-0 card-body"
        >
          <div
            className="card_info_section card"
            onClick={[Function]}
          >
            <div
              className="card_info_title card-title"
            >
              test6
            </div>
            <div
              className="p-0 card-body"
            >
              <p
                className="card_info_description"
              >
                description6
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
